test/factory/ZoraCreator1155Factory_Fork.t.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "forge-std/Test.sol";
+import {IZoraCreator1155Factory} from "../../src/interfaces/IZoraCreator1155Factory.sol";
+import {IZoraCreator1155} from "../../src/interfaces/IZoraCreator1155.sol";
+import {IMinter1155} from "../../src/interfaces/IMinter1155.sol";
+import {IOwnable} from "../../src/interfaces/IOwnable.sol";
+import {ICreatorRoyaltiesControl} from "../../src/interfaces/ICreatorRoyaltiesControl.sol";
+import {MockContractMetadata} from "../mock/MockContractMetadata.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {ZoraCreatorFixedPriceSaleStrategy} from "../../src/minters/fixed-price/ZoraCreatorFixedPriceSaleStrategy.sol";
+import {MintFeeManager} from "../../src/fee/MintFeeManager.sol";
+
+import {ChainConfig, Deployment} from "../../script/ZoraDeployerBase.sol";
+
+abstract contract ForkTesterBase is Test {
+    using stdJson for string;
+
+    /// @notice ChainID convenience getter
+    /// @return id chainId
+    function chainId() internal view returns (uint256 id) {
+        return block.chainid;
+    }
+
+    ///
+    // These are the JSON key constants to standardize writing and reading configuration
+    ///
+
+    string constant FACTORY_OWNER = "FACTORY_OWNER";
+    string constant MINT_FEE_AMOUNT = "MINT_FEE_AMOUNT";
+    string constant MINT_FEE_RECIPIENT = "MINT_FEE_RECIPIENT";
+
+    string constant FIXED_PRICE_SALE_STRATEGY = "FIXED_PRICE_SALE_STRATEGY";
+    string constant MERKLE_MINT_SALE_STRATEGY = "MERKLE_MINT_SALE_STRATEGY";
+    string constant REDEEM_MINTER_FACTORY = "REDEEM_MINTER_FACTORY";
+    string constant CONTRACT_1155_IMPL = "CONTRACT_1155_IMPL";
+    string constant FACTORY_IMPL = "FACTORY_IMPL";
+    string constant FACTORY_PROXY = "FACTORY_PROXY";
+
+    /// @notice Return a prefixed key for reading with a ".".
+    /// @param key key to prefix
+    /// @return prefixed key
+    function getKeyPrefix(string memory key) internal pure returns (string memory) {
+        return string.concat(".", key);
+    }
+
+    /// @notice Returns the chain configuration struct from the JSON configuration file
+    /// @return chainConfig structure
+    function getChainConfig() internal view returns (ChainConfig memory chainConfig) {
+        string memory json = vm.readFile(string.concat("chainConfigs/", Strings.toString(chainId()), ".json"));
```

Comment:
can there be a shared lib for this btw tests and deploys?

Reply:
yeah that would be ideal. i was just having a hard time cause in test, it inherits from `Test`, but in deploys, it inherits from `Script` - let me try to make it shared.

Reply:
done

Reply:
great!